PKCS12.xs

```diff
@@ -338,22 +338,32 @@ new(class)
 Crypt::OpenSSL::PKCS12
 new_from_string(class, string)
   SV  *class
-  char *string
+  SV  *string
 
   ALIAS:
   new_from_file = 1
 
   PREINIT:
   BIO *bio;
+  STRLEN str_len;
+  char *str_ptr;
 
   CODE:
 
-  if (!string) croak("PKCS12_new_from: No string or file was passed.");
+  SvGETMAGIC(string);
+
+  if (SvPOKp(string) || SvNOKp(string) || SvIOKp(string)) {
+    str_ptr = SvPV(string, str_len);
```

Comment:
Should this be `SvPVbyte` instead of `SvPV`

See: "[Perl’s SvPV Menace](https://dev.to/fgasper/perl-s-svpv-menace-5515)"

Reply:
@jonasbn Seems that's fine as we deal with octets input only, and never get back to Perl with input.

Below I improved input for `BIO_new_mem_buf()` in case someone accidentally uses UTF-8 encoded string as input. We know that this buffer must be octets.

For filename, both UTF-8 and non UTF-8 strings are allowed in `BIO_new_file()` (it uses `fopen()` internally). So letting `fopen()` consider if input is good.